editoast/openapi.yaml

```diff
@@ -2060,6 +2060,116 @@ components:
       - route_aspects
       - signals
       - mechanical_energy_consumed
+    SingleSimulationRequest:
+      properties:
+        allowances:
+          default: []
+          items:
+            $ref: '#/components/schemas/Allowance'
+          type: array
+        comfort:
+          $ref: '#/components/schemas/Comfort'
+        electrical_profile_set_id:
+          type: integer
+        initial_speed:
+          default: 0
+          format: float
+          type: number
+        options:
+          allOf:
+          - $ref: '#/components/schemas/TrainScheduleOptions'
+          nullable: true
+        path_id:
+          type: integer
+        power_restriction_ranges:
+          description: A list of ranges along the train path where power restrictions apply.
+          items:
+            $ref: '#/components/schemas/PowerRestrictionRange'
+          nullable: true
+          type: array
+        rolling_stock_id:
+          type: integer
+        scheduled_points:
+          default: []
+          items:
+            properties:
+              path_offset:
+                description: Offset in meters from the start of the path at which the train must be.
+                format: float
+                type: number
+              time:
+                description: Time in seconds (elapsed since the train's departure) at which the train must be.
+                format: float
+                type: number
+            required:
+            - path_offset
+            - time
+            type: object
+          type: array
+        stops:
+          default: []
+          items:
+            properties:
+              duration:
+                format: float
+                type: number
+              location:
+                $ref: '#/components/schemas/TrackLocation'
+              position:
```

Comment:
There might be an error here, when providing both a location and a position Core returns this error but works fine with one or the other:

```
{
    "status": 400,
    "type": "editoast:coreclient:Forward",
    "context": {
        "core_error": {
            "cause": null,
            "message": "Train stop must specify exactly one of position or location",
            "trace": null,
            "type": "invalid_schedule"
        },
        "status": 400,
        "url": "http://localhost:8080/standalone_simulation"
    },
    "message": "Train stop must specify exactly one of position or location"
}
```

Edit: `TrainStop` has a comment explaining this but it is not structurally ensured. **_If it doesn't involves too much changes_**, could you turn it into an enum? Thanks 😁

Reply:
Again, this will be handled in #5457